### DISCUSSION TITLE
Airflow DB Downgrade: Can't locate revision identified by <revision>

### DISCUSSION THREAD

**Discussion by lifayt** [apache/airflow#35357] [Q&A] [answered]
Hello,

I have a multi-cluster airflow environment on kubernetes, where all of our clusters are created by using the [airflow helm chart](https://airflow.apache.org/docs/helm-chart/stable/index.html). Unfortunately, we ran into an issue with permissions being cleared after upgrading to 2.6.2 (link [here](https://github.com/apache/airflow/discussions/32187)) and are trying to downgrade to 2.6.1 (if this fails, we would have to move forward to 2.7.1 latest).

At first, I tried rolling back our database by going into one of the airflow pods and running `airflow db downgrade --to-version=2.6.1` there, but it turns out the way our environment is set up the individual airflow pods don't get administrative credentials to our RDS database, so the only way to run the environment downgrade was during the helm deploy process. 

I changed the helm chart so that the migration pod ran a _downgrade_ rather than a db _upgrade_, but when the migration pod runs, it spits out the following error: 

```

alembic.script.revision.ResolutionError: No such revision or branch 'c804e5c76e3e'
...
alembic.util.exc.CommandError: Can't locate revision identified by 'c804e5c76e3e'
```

I looked up the revisions on the [reference page ](https://airflow.apache.org/docs/apache-airflow/stable/migrations-ref.html), and it appears that `c804e5c76e3e` is the reference to 2.6.2 so that kind of makes sense, but other than that I'm a little bit at a loss with regards to how to proceed from here. Any suggestions on how to resolve this state would be great, thanks!

**Comment by potiuk** [comment, marked as answer]
I think the problem is with changing to downgrade. I presume you set Airflow to 2.6.1 and attempted to run downgrade to 2.6.1. This won't (and can't) work because  Airflow 2.6.2 has absolutely no knowledg how to downgrade from 2.6.2. When Airlfow 2.6.1 was released, there was no 2.6.2 yet, so there is absolutely no way Airflow 2.6.1 can know how to move from 2.6.2 to 2.6.1 - only Airflow 2.6.2 and above know anything anout changes implemented in 2.6.2. So if you want to downgrade from 2.6.2 to something below - you shoudl use at least Airflow 2.6.2 to do so.

My recommendation is not to modify the chart - it has not been foreseen for those kinds of operations. Using Helm chart to manage changes to your databse and downgrade things is likely bad idea. The only thing the chart supports is to make sure database is upgraded to the right version from previous one when you had it - but this is merely `let's make sure this is "my" version`. 

For all the other things you should get access to regular terminal and console where you can run airflow migrate commands with admin access. Have Airlfow 2.6.2+  installed there (if your DB is in 2.6.2 version) and manually run the right commands. This is what the CLI commands of Airflow are for, so that you can use them to manage your instance and you need to make sure you have the right access to do so. Start a container in your cluster - configured with the DB access and right "Admin" credentials if you need to. Or you can run it on any machine that you can configure to access your DB. 

Our chart Helm does not give you such admin console out-of-the box - you need configure your admin console on your own.

**Reply by potiuk** [reply]
(of course you can also exec to any running container inside the chart - providing that your DB problem with admin access is soled).

**Reply by lifayt** [reply]
Thanks for responding to this - rolling forward to 2.7.1 was a functional solution for us, so that is what we did. I appreciate the detailed response.
